static-build/scripts/kata-configure-docker.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Description: Script to configure Docker for the static
+#   version of Kata Containers.
+
+[ -z "${DEBUG}" ] || set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+
+docker_config_file="${docker_config_file:-/etc/docker/daemon.json}"
+
+# The static version of Kata Containers is entirely contained within
+# this directory.
+readonly static_base_dir="/opt/kata"
+
+# Path to runtime in static archive file
+readonly runtime_path="${static_base_dir}/bin/kata-runtime"
+
+die()
+{
+	local msg="$*"
+	echo >&2 "ERROR: $msg"
+	exit 1
+}
+
+info()
+{
+	local msg="$*"
+	echo >&2 "INFO: $msg"
+}
+
+configure_docker()
+{
+	local file="$1"
+	[ -z "$file" ] && die "need file"
+
+	if [ -e "$docker_config_file" ]
+	then
+		local today=$(date '+%Y-%m-%d')
+		local backup="${docker_config_file}.${today}"
+
+		info "Backing up original Docker config file '$docker_config_file' to '$backup'"
+
+		sudo cp "${docker_config_file}" "${docker_config_file}.${today}"
+	fi
+
+	local config_files=$(tar tvf "$file" |\
+		grep "/configuration-.*\.toml" |\
+		grep -v -- '->' |\
+		awk '{print $NF}' |\
+		sed 's/^\.//g' || true)
+
+	[ -z "$config_files" ] && die "cannot find any configuration files in '$file'"
+
+	local config
+	local -a runtimes
+
+	for config in $(echo "$config_files" | tr '\n' ' ')
+	do
+		local runtime
+		runtime=$(echo "$config" |\
+			awk -F \/ '{print $NF}' |\
+			sed -e 's/configuration/kata/g' -e 's/\.toml//g')
+
+		runtimes+=("$runtime")
+
+		local result
+		result=$(cat "$docker_config_file" |\
+			jq \
+			--arg config "$config" \
+			--arg runtime "$runtime" \
+			--arg runtime_path "$runtime_path" \
+			'.runtimes[$runtime] = {path: $runtime_path, "runtimeArgs": ["--kata-config", $config]}')
+
+		echo "$result" > "$docker_config_file"
+	done
+
+	info "Validating $docker_config_file"
+
+	jq -S . "$docker_config_file" &>/dev/null
+
+	info "Restarting Docker to apply new configuration"
+
+	$chronic sudo systemctl restart docker
+
+	info "Docker configured for the following additional runtimes: ${runtimes[@]}"
+}
+
+setup()
+{
+	source "/etc/os-release" || source "/usr/lib/os-release"
+
+	# Used to manipulate $docker_config_file
+	local pkg="jq"
+
+	case "$ID" in
+		opensuse*) distro="opensuse" ;;
+		*)         distro="$ID" ;;
+	esac
+
+	# Use chronic(1) if available
+	chronic=
+	command -v chronic && chronic=chronic
+
+	if command -v "$pkg" &>/dev/null
+	then
+		return 0
+	fi
+
+	info "Installing package $pkg"
+
+	case "$distro" in
+		centos|rhel) $chronic sudo -E yum -y install "$pkg" ;;
+		debian|ubuntu) $chronic sudo -E apt -y install "$pkg" ;;
+		fedora) $chronic sudo -E dnf -y install "$pkg" ;;
+		opensuse|sles) $chronic sudo -E zypper -y install "$pkg" ;;
+		*) die "cannot handle distro: '$distro'" ;;
```

Comment:
nice but feels like it breaks a bit t he static philosophy,  on one hand I just love the way to handle the json here, but in other we are installing packages, it is not harmful and may be common is part of some systems. Well at least you check if yq is already installed  before,  instead of fail with:
`cannot handle distro` could you fail saying that  `yq is not installed.`

Reply:
Agreed - branch updated.